qmf/src/plugins/messageservices/imap/imapstrategy.cpp

```diff
@@ -155,13 +155,14 @@ bool updateMessagesMetaData(ImapStrategyContextBase *context,
                                    const QMailMessageKey &storedKey, 
                                    const QMailMessageKey &unseenKey, 
                                    const QMailMessageKey &seenKey,
+                                   const QMailMessageKey &flaggedAsDeleted,
                                    const QMailMessageKey &flaggedKey,
                                    const QMailMessageKey &unreadElsewhereKey,
                                    const QMailMessageKey &importantElsewhereKey,
                                    const QMailMessageKey &unavailableKey)
 {
     bool result = true;
-    QMailMessageKey reportedKey(seenKey | unseenKey);
+    QMailMessageKey reportedKey((seenKey | unseenKey)  & ~flaggedAsDeleted);
```

Comment:
Tested and working fine for me, I don't like this protocol behavior much, but some users complained a lot and they are right we are not following the protocol,
From user PoV I don't understand why would you want to have these emails around accumulating in your folders, in Thunderbird case expunge is never called unless you enable a extra option to do it on exit, leaving a lot of cruft until some other client calls expunge, to note that many of the major providers move the messages themselfs in the server side when deleted flag is issued from a client, so this case can never happen(e.g Gmail, Outlook, ...)